src/app/modules/forms/ix-forms/services/ix-form.service.ts

```diff
@@ -13,20 +17,79 @@ export class IxFormService {
     return this.getControls().map((ctrl) => ctrl.name);
   }
 
+  private getControlsOptions(query = ''): Option[] {
```

Comment:
1. I don't think `IxFormService` should care about searching within set of controls. Seems like a job for the component where it's needed.
2. What is `elementsWithIds`?
3. The code in two blocks is similar.